ptp/writer.go

```diff
@@ -57,9 +99,18 @@ type writerState struct {
 	fanSpeedsSeen    map[int]bool
 	temperaturesSeen map[float32]bool
 	layerHeightsSeen map[float32]bool
+
+	// keep track of seen max/min coordinates of print
+	boundingBox BoundingBox
 }
 
 func getStartingWriterState(initialExtrusionWidth, initialLayerHeight, zOffset float32) writerState {
+	negInf := float32(math.Inf(-1))
+	posInf := float32(math.Inf(1))
+	boundingBox := BoundingBox{
+		Min: Point{X: posInf, Y: posInf, Z: posInf},
+		Max: Point{X: negInf, Y: negInf, Z: negInf},
+	}
```

Comment:
It might be handy to make a "constructor function" that handles this for you (and would also encapsulate the `negInf` and `posInf` variables; by convention, it'd be called

```go
func MakeBoundingBox() BoundingBox {
  // ...
}
```